### ISSUE TITLE
Cannot edit my password if the input is small

### ISSUE THREAD

**Issue by geowa4** [wyantb/HashMask#26]
I have a password field that is only 94px wide. The HashMask canvas is currently taking up the entire input area so I cannot click on the password field to type.

I can tab to the password field though so I'm not blocked.

**Comment by deriachai** [comment]
Is this a public site we can test on?

**Comment by geowa4** [comment]
The site I noticed it on was a work product, but it should be easy enough to duplicate. I remembered one other thing about this. Chrome had the password saved so it was pre-filled. I went to change it, but I couldn't click on it, because the canvas was in the way.

**Comment by nickserv** [comment]
I made a test page for this here: http://wyantb.github.com/HashMask/small_password_field.html

The password field is autofocused by default, and I can type in the field as usual (though HashMask is in the way and is actually wider than the password field itself). Whenever I click on the password field (it's exactly 94px wide), it either deselects the password field and makes HashMask go away, or it reselects the password field and makes HashMask appear on top of it again.

TL;DR I'm not getting the same issue here. It works for me, despite it looking bad visually. I'm running a developer build of Chromium 18.0.1025.162, which should have very very similar behavior to Chrome (unless something updated). If someone with a normal release Chrome install could check out that test page, that would be nice.

**Comment by geowa4** [comment]
I get what you describe on that test page. On this internal site, it's a pre-filled login form. When I click on the canvas, it doesn't go away, but it does on your test page. Also, I'm on Chrome 20 (Linux).

**Comment by wyantb** [comment]
Hmm, odd that it's not going away.  Normally any unfocusing of the password field will make it hide, and clicking on the HasMask generally unfocuses the PW field.  In any case, could probably add specific code to hide it on click (I removed this at some point), and more importantly, let it get smaller in some cases (minimum 50px, maybe?  I don't know at what point it stops being legible).

**Comment by geowa4** [comment]
Ok, more details. When the sparkline first shows up in the pre-filled password field, I can click all I want and it won't go away. However, once I find a way to focus on the password field, clicking on the sparkline behaves as it does in the test page.

If you just make clicking on the sparkline focus the password field, I'd be happy. I'll gladly contribute that fix tonight if you want.

**Comment by geowa4** [comment]
I wouldn't resize it, because then the image will be different from what people remember.

**Comment by wyantb** [comment]
Have you had any problems identifying the same password in different locations with different heights?  That does change on different websites now, it's only width that doesn't change.

**Comment by wyantb** [comment]
I'll look towards doing this for r1.1, probably Monday, assuming no one beats me to it.

**Comment by deriachai** [comment]
Sparkline now takes the min of 100 pixels, or half of the input field. Works wonders on the 94 px one and is still easily recognizable.

Test and package in 1.1

**Comment by wyantb** [comment]
Just in case, I now also make sure to hide the sparkline when you click on it.  So this should be pretty decent in most cases.  

If you care to check the behavior before we release 1.1, chrome can load unpacked extensions, instructions [here](http://code.google.com/chrome/extensions/getstarted.html#load).  Not pandering, just not sure if you were aware or not.

**Comment by wyantb** [comment]
Addressed in 1.1, pushed to Web Store.  Should be available now.
